lake/branch.go

```diff
@@ -116,10 +116,19 @@ func (b *Branch) Delete(ctx context.Context, ids []ksuid.KSUID, author, message
 		if err != nil {
 			return nil, err
 		}
+		var objects []data.Object
```

Comment:
Nit: No point in copying these.
```suggestion
		var objects []*data.Object
```